lib/IManifest.ts

```diff
@@ -35,7 +36,7 @@ export async function manifestFromResource(testCaseHandlers: {[uri: string]: ITe
                 manifestFromSpecificationResource(testCaseHandlers, options, specificationResource) }))))) : null,
     subManifests: await Promise.all<IManifest>([].concat.apply([],
       resource.properties.include.map((includeList: Resource) => includeList.list
-        .map(manifestFromResource.bind(null, testCaseHandlers, options))))),
+        .map(resource => new ManifestLoader({ testCaseHandlers }).from(resource.value, options))))),
```

Comment:
@rubensworks - everything is working as I would now expect when I'm running the spec tests on SPARQL.js.

Without this change the test suites of nested submanifests were not run. This change also causes 3 failures in the test suite; as the test suite is now trying to fetch non-existent manifests when previously the fetch function was not invoked.

I believe that this is actually the behavior that is wanted, and that the tests should be fixed. But since I'm not overly familiar with this repository I would appreciate if you could confirm my logic is correct before I forge ahead and mock those tests.

Reply:
It's been a while since I worked in this code, but if I remember correctly, the helpers inside `IManifest` were not supposed to be aware of `ManifestLoader`, only the other way around (to avoid cycles in the architecture).

I suspect there may be a mismatch in how (sub)manifests are structured in the RDF-star test suite versus how its currently implemented here.
So we may to loosen the implementation here to handle that case, which should be possible without directly invoking `ManifestLoader` I think.

If I remember correctly, submanifests are also used in the SPARQL 1.1 test suite (as used in comunica).
Perhaps this can help in identifying the mismatch?

Reply:
Ah so it seems that the difference is recursive use of `mf:include` in the `-star` spec (the core manifest includes references to manifests that themselves have `mf:include`); so it seems the cleaner solution is to make this code segment recursive. https://github.com/rubensworks/rdf-test-suite.js/blob/5b795e0c8e055c8a6b6ec0306c6d7f65a7e21436/lib/ManifestLoader.ts#L55-L62

Reply:
@rubensworks so the problem is that the root manifest does `mf:include` on uris with the `.ttl` extension (see https://w3c.github.io/rdf-star/tests/manifest.ttl); however the manifest within the document that is traversed to _does not_ contain that extension (see https://w3c.github.io/rdf-star/tests/turtle/syntax/manifest.ttl).

On the other hand https://github.com/rubensworks/rdf-test-suite.js/blob/5b795e0c8e055c8a6b6ec0306c6d7f65a7e21436/lib/IManifest.ts#L37-L38 assumes they have the same URI

Reply:
This is the same problem that is discussed in https://github.com/w3c/rdf-star/issues/269#issuecomment-1186578380 to

Reply:
IMO this is poor practise for linked data publishing. However https://github.com/w3c/rdf-star/issues/269#issuecomment-1186578380 doesn't actually specify whether the URI in the entries list is the URL at which the manifest entries should be located or whether it is the IRI of the manifest itself.

Reply:
Hrm, that's annoying indeed. Let's report it (again?), and add a hardcoded workaround for this case?